rbhl/api.py

```diff
@@ -0,0 +1,166 @@
+from collections import defaultdict
+from decimal import Decimal
+from opal.core.views import json_response
+from opal.core.api import LoginRequiredViewset, episode_from_pk
+from rbhl.models import Demographics
+from opal.core.api import OPALRouter
+
+
+class PeakFlowGraphData(LoginRequiredViewset):
+    base_name = "peak_flow_graph_data"
+
+    def day_to_dict(self, peak_flow_day, demographics, pef):
+        aggregates = peak_flow_day.get_min_max_variability_completeness()
+
+        result = {
+            "note": peak_flow_day.note,
+            "treatment_taken": peak_flow_day.treatment_taken,
+            "day_num": peak_flow_day.day_num,
+            "date": peak_flow_day.date,
+            "work_day": peak_flow_day.work_day,
+            "pef_flow": pef
+        }
+
+        if not aggregates:
+            return result
+        else:
+            (
+                min_flow, max_flow, mean_flow, variabilty, completeness
+            ) = aggregates
+
+        result["min_flow"] = min_flow
+        result["mean_flow"] = mean_flow
+        result["max_flow"] = max_flow
+        result["variabilty"] = variabilty
+        result["completeness"] = completeness
+        return result
+
+    def get_completeness(self, day_dicts):
+        """
+        A complete day is every day where the patient has
+        added at least 6 entries.
+
+        If they miss a day, that's a day counted as incomplete
+        """
+
+        all_total_days = [i["day_num"] for i in day_dicts if "min_flow" in i]
+
+        if not all_total_days:
+            return
+
+        total_days = max(all_total_days)
+
+        completed_days = len(
+            [i for i in day_dicts if "completeness" in i and i["completeness"]]
+        )
+        if total_days:
+            completeness = Decimal(completed_days)/Decimal(total_days)
+        else:
+            completeness = 0
+
+        return round(completeness) * 100
+
+    def get_overrall_mean(self, day_dicts):
+        means = [i["mean_flow"] for i in day_dicts if "mean_flow" in i]
+        if means:
+            return round(sum(means)/len(means))
```

Comment:
When I read 'overall mean' I think you've taken the mean of all points not the mean of means....